### DISCUSSION TITLE
Is it possible to set a default metadata_public for all new users?

### DISCUSSION THREAD

**Discussion by knpwrs** [ory/kratos#2844] [Q&A]
I can see that it is possible to set `metadata_public` for new social sign-in using jsonnet data mapping: https://www.ory.sh/docs/kratos/social-signin/data-mapping

Does anything similar exist for non-social login? I'm using a field in `metadata_public` to represent a user's role. I would like to set all user's `metadata_public.role` to `'user'` by default.

**Comment by vinckr** [comment]
Hello Ken, 

something like this in the identity schema?

```
    "metadata_public": {
      "role": {
        "type": "string",
        "title": "Preferred Language",
        "default": "user",
        "enum": [
          "user",
          "admin",
          "superadmin"
        ]
      }
    }
```

You can already set it like this, the default does not work as expected however. You can add an API call after signup (or hook post signup) to add the role to the user. 
The way you proposed in https://github.com/ory/kratos/issues/2848#issuecomment-1295376534 would be neat, I am not sure if you can do that...

Maybe a good next step is to create an feature request issue for this. I will ask the team to take a look as well.
